continuous_integration/scripts/install.sh

```diff
@@ -23,6 +23,11 @@ if [[ ${UPSTREAM_DEV} ]]; then
         git+https://github.com/dask/distributed
 fi
 
+# Try nightly version of pyarrow
+conda uninstall --force pyarrow
+python -m pip install --extra-index-url https://pypi.fury.io/arrow-nightlies/ \
+    --pre pyarrow
```

Comment:
Could also put this in the `if [[ ${UPSTREAM_DEV} ]]` block above, and then actually merge this?